src/cmd/flux-job.c

```diff
@@ -2827,6 +2835,64 @@ int cmd_wait (optparse_t *p, int argc, char **argv)
     return (rc);
 }
 
+int cmd_annotate (optparse_t *p, int argc, char **argv)
+{
+    int optindex = optparse_option_index (p);
+    flux_t *h;
+    flux_jobid_t id;
+    const char *key;
+    const char *value;
+    void *valbuf = NULL;
+    json_t *v;
+    json_error_t error;
+    json_t *a;
+    flux_future_t *f;
+
+    if ((argc - optindex) != 3) {
+        optparse_print_usage (p);
+        exit (1);
+    }
+    if (!(h = flux_open (NULL, 0)))
+        log_err_exit ("flux_open");
+
+    id = parse_arg_unsigned (argv[optindex], "id");
```

Comment:
job ID's should now be parsed with `parse_jobid()` to support F58 and other representations.